bliss/backbone.py

```diff
@@ -0,0 +1,135 @@
+        self.num_imgs = n_imgs
+        self.model, self.save = parse_model(
+            deepcopy(self.yaml), ch=[ch], n_imgs=self.num_imgs
+        )  # model, savelist
+        self.names = [str(i) for i in range(self.yaml["nc"])]  # default names
+        self.inplace = self.yaml.get("inplace", True)
+        # Build strides, anchors
+        m = self.model[-1]  # Detect()
+        if isinstance(m, (Detect)):
+            s = 256  # 2x min stride
+            m.inplace = self.inplace
+            forward = lambda x: self.forward(x)  # pylint: disable=C3001, W0108  # noqa: E731
+
+            # NOTE: 5-D input
+            d = self.num_imgs  # set d = 2 in 5-band case
+            m.stride = torch.tensor([s / x.shape[-2] for x in forward(torch.zeros(1, ch, d, s, s))])
+
+            check_anchor_order(m)
+            m.anchors /= m.stride.view(-1, 1, 1)
+            self.stride = m.stride
+            self._initialize_biases()  # only run once
+
+    def forward(self, x, profile=False, visualize=False):
+        # pylint: disable=W0237
+        return self._forward_once(x, profile, visualize)  # single-scale inference, train
+
+    def _initialize_biases(self, cf=None):  # initialize biases into Detect(), cf is class frequency
+        # https://arxiv.org/abs/1708.02002 section 3.3
+        m = self.model[-1]  # Detect() module
+        for mi, s in zip(m.m, m.stride):  # from
+            b = mi.bias.view(m.na, -1)  # conv.bias(255) to (3,85)
+            b.data[:, 4] += math.log(8 / (640 / s) ** 2)  # obj (8 objects per 640 image)
+            b.data[:, 5 : 5 + m.nc] += (
+                math.log(0.6 / (m.nc - 0.99999)) if cf is None else torch.log(cf / cf.sum())
+            )  # cls
+            mi.bias = torch.nn.Parameter(b.view(-1), requires_grad=True)
+
+
+def parse_model(d, ch, n_imgs):  # model_dict, input_channels(3)
+    # pylint: disable=W0123, R0912
+    # Parse a YOLOv5 model.yaml dictionary
+    anchors, nc, gd, gw, act = (
+        d["anchors"],
+        d["nc"],
+        d["depth_multiple"],
+        d["width_multiple"],
+        d.get("activation"),
+    )
+    if act:
+        Conv.default_act = eval(  # noqa: S307, WPS421
+            act
+        )  # redefine default activation, i.e. Conv.default_act = nn.SiLU()
+    na = (len(anchors[0]) // 2) if isinstance(anchors, list) else anchors  # number of anchors
+    no = na * (nc + 5)  # number of outputs = anchors * (classes + 5)
+
+    layers, save, c2 = [], [], ch[-1]  # layers, savelist, ch out
+    for i, (f, n, m, args) in enumerate(d["backbone"] + d["head"]):  # from, number, module, args
+        m = eval(m) if isinstance(m, str) else m  # eval strings  # noqa: S307, WPS421
+        for j, a in enumerate(args):
+            with contextlib.suppress(NameError):
+                args[j] = eval(a) if isinstance(a, str) else a  # eval strings  # noqa: S307, WPS421
+
+        n = (  # noqa: WPS429, WPS120, F841 # pylint: disable=W0612
```

Comment:
Would you disable pylint and flake8 for the whole file rather than going line-by-line, since there are so many violoations? And perhaps add a comment to the top of the file explaining that this isn't our code, it's yolo5v code, but we're including it so we can use conv3d.